fuel-vm/src/tests/metadata.rs

```diff
@@ -268,6 +268,53 @@ fn get_metadata_chain_id() {
     }
 }
 
+#[test]
+fn get_metadata_base_asset_id() {
```

Comment:
So, I'd say that the SUT of this test is `op::gm_args` and the condition is `GMArgs::BaseAssetId`. It's confusing since we can't call `gm_args` natively and instead we call `transact`, so you could also say that `transact` is the SUT in that sense.

Either way, I think the structure of the test should reflect what the SUT is. In this case I include it in a `gm_args_script` method, so it's a little indirect but cleans up the test in a good way IMO:

```
// given
let base_asset_id = AssetId::from([5; 32]);
let params = ConsensusParameters {
        base_asset_id,
        ..Default::default()
    };

// when
let arg = GMArgs::BaseAssetId;
let size = AssetId::LEN;
let script = gm_args_script(arg, size);

let logged_data = run_script_and_get_logged_data(params, script);

// then
let expected = params.base_asset_id.to_bytes();
let actual = logged_data;
assert!(expected, actual);
```

So the name of the test could be something like `gm_args__base_asset_id_retrieves_value_in_params`

Reply:
I think doing that would make the test case less readable. Note how this is already an integration test, not a unit test, and it still has minimal indirection which is quite nice. I experimented a bit with something like `run_script_and_get_logged_data` and going that way seems to lead essentially duplicating `TransactionBuilder` if we want to make it reusable.